From 60256f28189c3d0650a78e737eb0ca4753478a4b Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Fri, 6 Apr 2018 04:15:45 +0000
Subject: upstream: relax checking of authorized_keys environment="..."

options to allow underscores in variable names (regression introduced in
7.7). bz2851, ok deraadt@

OpenBSD-Commit-ID: 69690ffe0c97ff393f2c76d25b4b3d2ed4e4ac9c

Original-Author: Damien Miller <djm@mindrot.org>
Origin: backport, http://anongit.mindrot.org/openssh.git/commit/?id=40f5f03544a07ebd2003b443d42e85cb51d94d59
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1771011
Last-Update: 2018-06-28
Signed-off-by: Christian Ehrhardt <christian.ehrhardt@canonical.com>

Patch-Name: upstream-relax-checking-of-authorized_keys-environme.patch
---
 auth-options.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/auth-options.c b/auth-options.c
index b528c197a..ef57ebf43 100644
--- a/auth-options.c
+++ b/auth-options.c
@@ -1,4 +1,4 @@
-/* $OpenBSD: auth-options.c,v 1.78 2018/03/14 05:35:40 djm Exp $ */
+/* $OpenBSD: auth-options.c,v 1.79 2018/04/06 04:15:45 djm Exp $ */
 /*
  * Copyright (c) 2018 Damien Miller <djm@mindrot.org>
  *
@@ -394,7 +394,7 @@ sshauthopt_parse(const char *opts, const char **errstrp)
 				goto fail;
 			}
 			for (cp = opt; cp < tmp; cp++) {
-				if (!isalnum((u_char)*cp)) {
+				if (!isalnum((u_char)*cp) && *cp != '_') {
 					free(opt);
 					errstr = "invalid environment string";
 					goto fail;
